lib/ipc/ipc.go

```diff
@@ -1,11 +1,11 @@
 package ipc
 
-type IpcWriter interface {
```

Comment:
this looks a lot like https://pkg.go.dev/io#WriteCloser. and IPCReader like ReadCloser. would you want to have these be a named wrapper for those interfaces. 
Even if not, the implementations of Cleanup here now are printing errors rather than returning them which is probably not good practice long term. May want to specify that Cleanup returns an error so those functions can report errors in a helpful way.

Reply:
Lets make this another PR lol